File: maven-plugin/src/main/java/io/github/potjerodekool/openapi/maven/MavenLogger.java
package io.github.potjerodekool.openapi.maven;

import io.github.potjerodekool.openapi.log.LogLevel;
import io.github.potjerodekool.openapi.log.Logger;
import org.apache.maven.plugin.Mojo;

public class MavenLogger implements Logger {

    private final Mojo mojo;
    private final String name;

    public MavenLogger(final Mojo mojo,
                       final String name) {
        this.mojo = mojo;
        this.name = name;
    }

    @Override
    public void log(final LogLevel level, final String message, final Throwable exception) {
        final var logMessage = String.format("%s: %s", name, message);

        switch (level) {
            case SEVERE -> mojo.getLog().error(logMessage, exception);
            case WARNING -> mojo.getLog().warn(logMessage, exception);
            default -> mojo.getLog().info(logMessage, exception);
        }
    }
}

File: maven-plugin/src/main/kotlin/io/github/potjerodekool/openapi/maven/MavenLogger.kt
package io.github.potjerodekool.openapi.maven

import io.github.potjerodekool.openapi.common.log.LogLevel
import io.github.potjerodekool.openapi.common.log.Logger
import org.apache.maven.plugin.Mojo

class MavenLogger(
    private val mojo: Mojo,
    private val name: String
) : Logger {
    override fun log(level: LogLevel?, message: String?, exception: Throwable?) {
        val logMessage = String.format("%s: %s", name, message)

        when (level) {
            LogLevel.SEVERE -> mojo.log.error(logMessage, exception)
            LogLevel.WARNING -> mojo.log.warn(logMessage, exception)
            else -> mojo.log.info(logMessage, exception)
        }
    }
}
